rfc/ecs_e2e_tests.md

```diff
@@ -0,0 +1,371 @@
+
+
+## Prior Art and Alternatives
+
+In parallel to adding e2e tests, the ECS team is also working on `clouddriver` 
+integration tests. Integration tests are a better venue for more exhaustive 
+testing, especially for configurations that require few supporting resources. 
+Re: ECS server groups, test cases that do not require an artifact 
+(see **Test case #2**, above) or use a load balancer come to mind as good 
+first candidates for these tests, in addition to basic read opertions 
+on `/loadBalncers`, `/credentials`, etc.
+
+These kinds of tests also align with where the community wants to shift 
+the bulk of future testing, and it's more conducive to automation on PRs, 
+which would help detect problems before they even make it to `master`. 
+In an effort to make this a reality we've built a 
+[very early proof of concept integration test](https://github.com/allisaurus/clouddriver/tree/ecs-integ/clouddriver-ecs/src/integration) 
+for ECS which is triggered by a GitHub action and are working to get 
+it into a state that can run on developer forks.
+
+### Why not just integration tests?
+
+My ideal scenario would be to have one (or very few, strategically chosen) 
+e2e ECS test running to ensure the contracts between services are intact prior to 
+release, but with most permutations of the available atomic operations and 
+controllers run as integration tests on PR submission. 
+
+While we're actively iterating on the integration tests to make them more meaningful, 
+they still have major limitations (e.g., they require an actual AWS account) which 
+will take time to sort out. Community guidance re: what integration tests should 
+look like is also still missing, though conversations have started w/r/t requirements.
+
+So given the early days of integration tests and the existance of the 
+current e2e stack, our best option for contributing automated ECS tests 
+before the end of 2020 (only one release remains) seems to be by 
+adding to the existing e2e stack as described above. After that, 
+the state of available integration test code & infrastructure 
+will inform where additional test cases should reside. (Or, even later, if/when
+we can remove the e2e tests entirely)
+
+
+## Known Unknowns
+
+Some things may impact this proposal which I'm hoping the RFC process can clear up:
+
+### About the test design
+
+* Whether any Jenkins/e2e test environment configuration would be impacted 
+  by this change but is not immediately obvious given the [`buildtool`](https://github.com/spinnaker/buildtool) code.
+* Whether testing a provider that relies on another provider (e.g, AWS) 
+  might cause issues with an unknown (to me) authentication or test mechanism.
+* What's the best way to test any new configuration + test code prior to merging? 
+* Unknown/show-stopping restrictions on the AWS account currently used during tests.
+* How many regions are running the e2e tests, and would we need a unique ECR repository for each?
```

Comment:
I think they only run in one region, though that region comes from an environment variable.  It's currently set to `us-east-1` though there have been cases where we've moved things if a particular region was having issues and we didn't want the tests to fail.